lib/Sema/CSSolver.cpp

```diff
@@ -1116,8 +1116,7 @@ static bool debugConstraintSolverForTarget(
 /// diagnostic.
 static void maybeProduceFallbackDiagnostic(
     ConstraintSystem &cs, SolutionApplicationTarget target) {
-  if (cs.Options.contains(ConstraintSystemFlags::SubExpressionDiagnostics) ||
```

Comment:
Thats more a question, the flag `ConstraintSystemFlags::SubExpressionDiagnostics` is used for something else than CSDiag typecheck? I'm just asking because this maybe one thing else that can be removed :)